### DISCUSSION TITLE
How to check if a Livewire property has changed / is dirty?

### DISCUSSION THREAD

**Discussion by marbuser** [livewire/livewire#2289] [1. Help]
So I've got a Livewire application, and when a property is updated and is different than what it was previously, I would like to run a certain piece of code. (My "activity" logger.)

For context, I'm doing file uploads and would like to know when the user uploads something different and saves it. If the user just saves without updating the uploads, then I do not want the code to execute. If they do change it however, then I do want the code to execute. (To avoid duplicate/inaccruate "activity" being logged. If they didn't change the uploads then it shouldn't be logged.)

I was hoping I could just do something along the lines of;
`if ($this->isDirty('property'))`

But this doesn't exist within Livewire. I've tried searching around and source diving but can't really seem to find anything of this nature.

Not event really sure if this is possible with Livewire. Hopefully someone can provide some advice. 👍

**Comment by ju5t** [comment]
You could work around this by adding some logic in `updating`. This is one of the [Lifecycle Hooks](https://laravel-livewire.com/docs/2.x/lifecycle-hooks) in Livewire. It is passed a name (in dot notation if it's an array) and the new value. You could use this to fetch the current value and decide if they are different.

**Reply by ju5t** [reply]
For those reading this later on; this is a trait we're using to check if something has changed at all. It doesn't track specific properties yet, but it could if you made some changes.

```php
<?php

namespace App\Http\Livewire\Traits;

use Livewire\Component;

/** @mixin Component */
trait TrackDirtyProperties
{
    public bool $isDirty = false;

    /** @noinspection PhpUnused */
    public function updatedTrackDirtyProperties(): void
    {
        $this->isDirty = true;
    }

    public function cleanUp(): void
    {
        $this->isDirty = false;
    }
}
```
